### PR TITLE
added resizable column using react-split

### PR DESCRIPTION
**Pull request by MaseruKganyago**

### CODE REVIEW

File: shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+import { nanoid } from 'nanoid/non-secure';
+
+export interface IProps {
+  readOnly: boolean;
+  value?: object;
+  onChange?: any;
+}
+
+const EditableContext = React.createContext(null);
+const DragHandleContext = React.createContext(null);
```
Comment (Lukeybooi) [inline shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx:24]: Please add a space after the context for readability

File: shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+const EditableCell = ({ title, editable, children, dataIndex, record, handleSave, ...restProps }) => {
+  const [editing, setEditing] = useState(false);
+  const inputRef = useRef(null);
+  const form = useContext(EditableContext);
```
Comment (Lukeybooi) [inline shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx:29]: Please add a space here for readability

File: shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        }}
+        name={dataIndex}
+        rules={[
+          {
+            required: true,
+            message: `${title} is required.`,
+          },
+        ]}
+      >
+        {/* <Input ref={inputRef} onPressEnter={save} onBlur={save} /> */}
```
Comment (Lukeybooi) [inline shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx]: Please remove this

File: shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        record,
+        editable: col.editable,
+        dataIndex: col.dataIndex,
+        title: col.title,
+        handleSave: handleSaveCell,
+      }),
+    };
+  });
+
+  const getListStyle = (_isDraggingOver: boolean) => ({
```
Comment (Lukeybooi) [inline shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx]: There is no point of this function, please add functionality or remove it

File: shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+  border: isDragging ? '1px dashed #000' : 'none',
+  ...draggableStyle,
+});
+
+const DraggableBodyRowInner = ({ columns, className, style, ...restProps }) => {
+  const [form] = Form.useForm();
+
+  // function findIndex base on Table rowKey props and should always be a right array index
+  const rowKey = restProps['data-row-key'];
+  const index = columns.findIndex((x) => x.id === restProps['data-row-key']);
```
Comment (Lukeybooi) [inline shesha-reactjs/src/components/formDesigner/components/sizableColumns/sizableColumnList.tsx:102]: Please add space here

File: shesha-reactjs/src/components/formDesigner/components/sizableColumns/__tests__/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (Lukeybooi) [inline shesha-reactjs/src/components/formDesigner/components/sizableColumns/__tests__/index.test.tsx]: There's no point of this. Please remove this unit shallow unit test

Comment (MaseruKganyago) [reply shesha-reactjs/src/components/formDesigner/components/sizableColumns/__tests__/index.test.tsx]: Was added as to follow the standard of all components

**Review by Lukeybooi — APPROVED** [review]

**Review by Lukeybooi — APPROVED** [review]

**Review by Lukeybooi — APPROVED** [review]

**Pull request merged**